src/deep_compare.js

```diff
@@ -0,0 +1,22 @@
+export const deep_compare = (objectOne, objectTwo) => {
+  if (typeof objectOne !== 'object' || typeof objectTwo !== 'object') {
+    return false
+  }
+
+  let objectOneProps = Object.getOwnPropertyNames(objectOne)
+  let objectTwoProps = Object.getOwnPropertyNames(objectTwo)
```

Comment:
I am in favor of using `const` here since `objectOneProps` and `objectTwoProps` are never mutated and are not meant to be:
```suggestion
  const objectOneProps = Object.getOwnPropertyNames(objectOne)
  const objectTwoProps = Object.getOwnPropertyNames(objectTwo)
```